test/modules/standard/gmp/ferguson/gmp_dist_array.chpl

```diff
@@ -44,16 +41,27 @@ modulus.nextprime(modulus);
 writeln(modulus);
 sum.mod(sum, modulus);
 writeln(sum);
-delete modulus;
-delete sum;
+
 
 // Release the array elements.
+// TODO: remove this section once the records are being released correctly.
```

Comment:
Oh right.  That was dumb on my part.

Tough call.  I believe I would choose to let this test leak then
but I can understand someone taking the opposing point of
view.


In the original version of the app, the app was responsible for
memory management.  In the new version it’s not supposed
to be.


We have other leaks in other apps right now that are are due to
lower-level bugs.   In every case that I know about we’ve elected
to leave the leak in the App.


I know of one sort-of exception to this rule and was a leak in a
popular library that used inherited a leak.  We chose to modify
the library to cover the leak until we can fix the leak.

Mike







On Jul 28, 2016, at 2:46 PM, Laura Delaney <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/modules/standard/gmp/ferguson/gmp_dist_array.chpl<https://github.com/chapel-lang/chapel/pull/4243#discussion_r72709196>:

>
>  // Release the array elements.
> +// TODO: remove this section once the records are being released correctly.


I'm not adding manual_free to the BigInt type module, just to that one test so that it leaks less (it still leaks 2000 bytes of other data related to leaking the array elements).

If we would prefer to let the test leak everything, we could do that.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/chapel-lang/chapel/pull/4243/files/3bd5d074a99d2427218177f58af6019dc3d8c2be#r72709196>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACERpltxpu50b_M0W-Oi9NEoiKjto-9lks5qaSNTgaJpZM4JTwJ0>.